topology-machine/topomachine

```diff
@@ -187,41 +215,98 @@ def run_topology(config, dry_run, with_trace=False):
     for router in sorted(config['routers']):
         val = config['routers'][router]
         name = "%s%s" % (args.prefix, router)
-        cmd = ["docker", "run", "--privileged", "-d",
-               "--name", name,
-               "%svr-%s:%s" % (docker_registry, val["type"], val["version"])
-        ]
+        image = "%svr-%s:%s" % (docker_registry, val["type"], val["version"])
+
+        api_cmd = []
         if trace:
-            cmd.append(trace)
+            api_cmd.append(trace)
         if 'run_args' in val:
-            cmd.extend(val["run_args"].split())
+            api_cmd.extend(val["run_args"].split())
 
-        output,_ = run_command(["docker", "inspect", "--format", "{{.State.Running}}", name])
-        if output.strip() == "true":
-            output,_ = run_command(["docker", "inspect", "--format", "{{.State.Health.Status}}", name])
-            print("Container already running. Health: %s" % output.strip())
+        state_output = ''
+        try:
+            state_output = get_container_state_api(url, name)
+        except:
+            pass
```

Comment:
Do we really want to suppress error messages here? If something goes to shit, ISTM that it's better to just throw an exception, no?

Reply:
For example, if a container doesn't exist and we were looking for it's existence, perhaps this should be caught somewhere and returned more as a regular return value?